datafusion/optimizer/src/push_down_limit.rs

```diff
@@ -153,6 +153,36 @@ impl OptimizerRule for PushDownLimit {
                 subquery_alias.input = Arc::new(new_limit);
                 Ok(Transformed::yes(LogicalPlan::SubqueryAlias(subquery_alias)))
             }
+            LogicalPlan::Extension(extension_plan) => {
+                if !extension_plan.node.allows_limit_to_inputs() {
```

Comment:
`if extension_plan.node.allows_limit_to_inputs()` could be moved to match guard:
`LogicalPlan::Extension(extension_plan) if extension_plan.node.allows_limit_to_inputs()`

Reply:
Sure thing, thanks for review!